product_multi_barcode_supplierinfo/models/supplierinfo.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2022 ForgeFlow S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SupplierInfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    def fill_barcode_ids(self):
+        if self.barcode:
+            if self.product_id:
+                self.env["product.barcode"].create(
+                    {"name": self.barcode, "product_id": self.product_id.id}
+                )
+            else:
+                self.env["product.barcode"].create(
+                    {"name": self.barcode, "product_tmpl_id": self.product_tmpl_id.id}
+                )
+
+    @api.model
+    def create(self, vals):
+        records = super().create(vals)
+        for rec in records:
+            rec.fill_barcode_ids()
+        return records
+
+    def write(self, vals):
+        for rec in self:
+            if rec.barcode != vals.get("barcode"):
+                rec.unlink_product_barcode()
+        res = super().write(vals)
+        for rec in self:
+            rec.fill_barcode_ids()
```

Comment:
if you are writing a different field, I there is nothing that prevents to create a duplicated product.barcode, right?